common/persistence/dataInterfaces.go

```diff
@@ -810,6 +810,17 @@ type (
 		FailoverVersion   int64
 	}
 
+	// FlatCreateDomainRequest is a flattened version of CreateDomainRequest
+	FlatCreateDomainRequest struct {
```

Comment:
Rename this to DomainRow, and move it to sqlMetadataPersistence, as it is not of interest to the general implementation.